src/Nethermind/Nethermind.Evm/TransactionProcessing/TransactionProcessor.cs

```diff
@@ -93,278 +94,396 @@ public void Trace(Transaction transaction, BlockHeader block, ITxTracer txTracer
                 {
-                    TraceLogInvalidTx(transaction, "NONCE_OVERFLOW");
-                    QuickFail(transaction, block, txTracer, eip658NotEnabled, "nonce overflow");
-                    return;
+                    TraceLogInvalidTx(tx, "NONCE_OVERFLOW");
+                    QuickFail(tx, blk, spec, tracer, "nonce overflow");
+                    return false;
                 }
             }
 
-            if (transaction.IsAboveInitCode(spec))
+            if (tx.IsAboveInitCode(spec))
             {
-                TraceLogInvalidTx(transaction, $"CREATE_TRANSACTION_SIZE_EXCEEDS_MAX_INIT_CODE_SIZE {transaction.DataLength} > {spec.MaxInitCodeSize}");
-                QuickFail(transaction, block, txTracer, eip658NotEnabled, "EIP-3860 - transaction size over max init code size");
-                return;
+                TraceLogInvalidTx(tx, $"CREATE_TRANSACTION_SIZE_EXCEEDS_MAX_INIT_CODE_SIZE {tx.DataLength} > {spec.MaxInitCodeSize}");
+                QuickFail(tx, blk, spec, tracer, "EIP-3860 - transaction size over max init code size");
+                return false;
             }
 
-            long intrinsicGas = IntrinsicGasCalculator.Calculate(transaction, spec);
-            if (_logger.IsTrace) _logger.Trace($"Intrinsic gas calculated for {transaction.Hash}: " + intrinsicGas);
-
-            if (notSystemTransaction)
+            if (!tx.IsSystem())
             {
-                if (gasLimit < intrinsicGas)
+                if (tx.GasLimit < intrinsicGas)
                 {
-                    TraceLogInvalidTx(transaction, $"GAS_LIMIT_BELOW_INTRINSIC_GAS {gasLimit} < {intrinsicGas}");
-                    QuickFail(transaction, block, txTracer, eip658NotEnabled, "gas limit below intrinsic gas");
-                    return;
+                    TraceLogInvalidTx(tx, $"GAS_LIMIT_BELOW_INTRINSIC_GAS {tx.GasLimit} < {intrinsicGas}");
+                    QuickFail(tx, blk, spec, tracer, "gas limit below intrinsic gas");
+                    return false;
                 }
 
-                if (!noValidation && gasLimit > block.GasLimit - block.GasUsed)
+                if (validate && tx.GasLimit > blk.GasLimit - blk.GasUsed)
                 {
-                    TraceLogInvalidTx(transaction,
-                        $"BLOCK_GAS_LIMIT_EXCEEDED {gasLimit} > {block.GasLimit} - {block.GasUsed}");
-                    QuickFail(transaction, block, txTracer, eip658NotEnabled, "block gas limit exceeded");
-                    return;
+                    TraceLogInvalidTx(tx, $"BLOCK_GAS_LIMIT_EXCEEDED {tx.GasLimit} > {blk.GasLimit} - {blk.GasUsed}");
+                    QuickFail(tx, blk, spec, tracer, "block gas limit exceeded");
+                    return false;
                 }
             }
 
-            if (!_worldState.AccountExists(caller))
+            return true;
+        }
+
+        // TODO Should we remove this already
```

Comment:
proposition: to resolve this ToDo we could add _logger with warning here. and observe on nodes?

Reply:
done